abb_bringup/launch/abb_moveit2.launch.py

```diff
@@ -0,0 +1,182 @@
+from ament_index_python.packages import get_package_share_directory
+from launch import LaunchDescription
+from launch.actions import DeclareLaunchArgument
+from launch.actions import OpaqueFunction
+from launch.conditions import IfCondition
+from launch.substitutions import Command, FindExecutable, LaunchConfiguration, PathJoinSubstitution
+from launch_ros.actions import Node
+from launch_ros.substitutions import FindPackageShare
+import os
+import yaml
+
+
+def load_yaml(package_name, file_path):
+    package_path = get_package_share_directory(package_name)
+    absolute_file_path = os.path.join(package_path, file_path)
+
+    try:
+        with open(absolute_file_path, "r") as file:
+            return yaml.safe_load(file)
+    except EnvironmentError:  # parent of IOError, OSError *and* WindowsError where available
+        return None
+
+def launch_setup(context, *args, **kwargs):
+    # Command-line arguments
+    robot_xacro_file = LaunchConfiguration("robot_xacro_file")
+    support_package = LaunchConfiguration("support_package")
+    moveit_config_package = LaunchConfiguration("moveit_config_package")
+    moveit_config_file = LaunchConfiguration("moveit_config_file")
+
+    # Planning context
+    robot_description_content = Command(
+        [
+            PathJoinSubstitution([FindExecutable(name="xacro")]),
+            " ",
+            PathJoinSubstitution([FindPackageShare(support_package), "urdf", robot_xacro_file]),
+        ]
+    )
+    robot_description = {"robot_description": robot_description_content}
+
+    robot_description_semantic_content = Command(
+        [
+            PathJoinSubstitution([FindExecutable(name="xacro")]),
+            " ",
+            PathJoinSubstitution(
+                [FindPackageShare(moveit_config_package), "config", moveit_config_file]
+            ),
+        ]
+    )
+    robot_description_semantic = {
+        "robot_description_semantic": robot_description_semantic_content
```

Comment:
```suggestion
        "robot_description_semantic": robot_description_semantic_content.perform(context)
```

Reply:
Not sure if there's a more ergonomic way to do this, but this seems to fix the issue for me...